server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -851,18 +849,24 @@ public void write(XContentBuilder b) throws IOException {
 
             if (ignoredValues != null && ignoredValues.isEmpty() == false) {
                 // Use an ordered map between field names and writer functions, to order writing by field name.
-                Map<String, CheckedConsumer<XContentBuilder, IOException>> orderedFields = new TreeMap<>();
+                Map<String, FieldWriter> orderedFields = new TreeMap<>();
                 for (IgnoredSourceFieldMapper.NameValue value : ignoredValues) {
-                    orderedFields.put(value.name(), value::write);
+                    var existing = orderedFields.get(value.name());
+                    if (existing == null) {
+                        orderedFields.put(value.name(), new FieldWriter.IgnoredSource(value));
+                    } else if (existing instanceof FieldWriter.IgnoredSourceWriter isw) {
+                        orderedFields.put(value.name(), isw.mergeWith(value));
+                    }
                 }
                 for (SourceLoader.SyntheticFieldLoader field : fields) {
                     if (field.hasValue()) {
                         // Skip if the field source is stored separately, to avoid double-printing.
-                        orderedFields.putIfAbsent(field.fieldName(), field::write);
+                        orderedFields.putIfAbsent(field.fieldName(), new FieldWriter.FieldLoader(field));
```

Comment:
This doesn't allow mixing some ignored values with regular values? E.g. what happens when we have an array with some malformed values?

Also, this keeps creating new FieldLoaders, even if they're not used. Use `computeIfAbsent` instead?

Reply:
Yes, we don't merge (keeping the current behavior). Most of the time it is not correct to merge right now since it will result in duplicate values. If we wanted to merge we would need a clear signal on when it is legal to do so and i am not sure what that would be.

When we need to mix valid values with ignored ones we create a dedicated stored field to store ignored values. However we do use ignored source directly to handle `ignore_above` in `flattened` field. I think that use case is broken with arrays. I'll file an issue.